library/waku_thread/waku_thread.nim

```diff
@@ -0,0 +1,188 @@
+
+{.pragma: exported, exportc, cdecl, raises: [].}
+{.pragma: callback, cdecl, raises: [], gcsafe.}
+{.passc: "-fPIC".}
+
+import
+  std/[json,sequtils,times,strformat,options,atomics,strutils,os]
+import
+  chronicles,
+  chronos,
+  stew/results,
+  stew/shims/net
+import
+  ../../../waku/common/enr/builder,
+  ../../../waku/v2/waku_enr/capabilities,
+  ../../../waku/v2/waku_enr/multiaddr,
+  ../../../waku/v2/waku_enr/sharding,
+  ../../../waku/v2/waku_core/message/message,
+  ../../../waku/v2/waku_core/topics/pubsub_topic,
+  ../../../waku/v2/node/peer_manager/peer_manager,
+  ../../../waku/v2/waku_core,
+  ../../../waku/v2/node/waku_node,
+  ../../../waku/v2/node/builder,
+  ../../../waku/v2/node/config,
+  ../../../waku/v2/waku_relay/protocol,
+  ../events/[json_error_event,json_message_event,json_base_event],
+  ../alloc,
+  ./config,
+  ./inter_thread_communication/request
+
+type
+  Context* = object
+    thread: Thread[(ptr Context)]
+    reqChannel: Channel[InterThreadRequest]
+    respChannel: Channel[Result[string, string]]
+    node: WakuNode
+
+var ctx {.threadvar.}: ptr Context
```

Comment:
instead of using a global here, it's better to return the context to the C api and ask them to pass it in to every function